bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/productteaser/ProductTeaser.java

```diff
@@ -61,23 +61,32 @@
     String getUrl();
 
     /**
-     * Returns the SKU of the product displayed by this {@code ProductTeaser}
-     * 
+     * Returns the SKU of the product displayed by this {@code ProductTeaser}. If the teaser displays a variant, then this method returns
+     * the SKU of the variant.
+     *
      * @return a String value representing the SKU
      */
     String getSku();
 
+    /**
```

Comment:
Why is this needed? The information is present in the sku and the format is also kown/fixed to `sku#variant` in many places. Can the consumer extract the "master product sku" instead from the sku and we save the redundancy here?

Reply:
The problem here is that the Product Teaser does not expose the `sku#variant` format. Also, no consumer is extracting the "master product SKU", everything is done at the model level. 

Some context: all the components that display products rely on the "master SKU" because you cannot search by variant SKU in Magento. The product teaser is the only one (apart from the `Product` component, but that's a special case) that retrieves the product directly by the SKU, so that's why it doesn't expose the `combinedSku` format.

Reply:
If any other place returns `sku#variant`, why doesn't the product teaser? We should fix that instead. Afaics only the data layer currently consumes this sku. 

I am -1 introducing this technical dept in combination with a major interface change. 

Lets better introduce a productteaser v2 that does the sku right.

Reply:
> If any other place returns sku#variant

No other places return `sku#variant`. All the models return the master product SKU, that's why I didn't have to make any changes in other models.